pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/immutable/EmptyIndexSegment.java

```diff
@@ -99,6 +101,11 @@ public GenericRow getRecord(int docId, GenericRow reuse) {
     throw new UnsupportedOperationException();
   }
 
+  @Override
+  public final Map<String, ColumnIndexContainer> getIndexContainerMap() {
+    return null;
```

Comment:
Return `Collections.emptyMap()` here? Or it could cause NPE while processing the map